js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,133 @@ export const ResourcesPolicySelect = ({
     [search]
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("common:unsavedChangesTitle"),
+    messageKey: t("common:unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
```

Comment:
There is no need for this to be an `async` function, as there is no Promise returned from it.
```suggestion
    onConfirm() {
```